cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -53,6 +54,7 @@ var (
 	argEtcdServer          = flag.String("etcd-server", "http://127.0.0.1:4001", "URL to etcd server")
 	argKubecfgFile         = flag.String("kubecfg_file", "", "Location of kubecfg file for access to kubernetes master service; --kube_master_url overrides the URL part of this; if neither this nor --kube_master_url are provided, defaults to service account tokens")
 	argKubeMasterURL       = flag.String("kube_master_url", "", "URL to reach kubernetes master. Env variables in this flag will be expanded.")
+	dnsLabelRegex, _       = regexp.Compile("^[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9]$")
```

Comment:
This is not right.  We don't want to allow upper-case and it is more complex than required.  You should just reference pkg/util/validation DNS1123LabelFmt: `regexp.Compile("^" + validation.DNS1123LabelFmt + "$")`